### DISCUSSION TITLE
IKEA VINDSTYRKA (E2112) VOC sensor device_class in Home Assistant

### DISCUSSION THREAD

**Discussion by tomaszn** [Koenkk/zigbee2mqtt#24428] [Devices]
IKEA VINDSTYRKA reports volatile organic compounds (VOC) values correctly in Zigbee2MQTT (Z2M). However, these measurements lack a device class in Home Assistant, preventing their inclusion in an area badge.

Issue #17208 was previously closed because the VOC sensor did not use the µg/m³ unit, which was required for setting the device class at that time.

However, Home Assistant now supports two device classes for VOC:

> volatile_organic_compounds: Concentration in µg/m
> volatile_organic_compounds_parts: Ratio in ppm or ppb

What changes are necessary to classify the VINDSTYRKA VOC measurements as `volatile_organic_compounds_parts` in Home Assistant?

**Comment by Koenkk** [comment]
What is the unit of `pm25`? Currently it is `µg/m³` and for `_parts` it needs to be `ppm` or `ppb` (CC: @sjorge)

**Reply by sjorge** [reply]
If IIRC it reports a raw µg/m³, but that was reverse engineered by comparing the output from the tradfri app vs what I was seeing on the mesh.

At the time of writing at least our values matched the tradfri app.

Don't own a hub to check again.

**Reply by tomaszn** [reply]
`pm25` is a different sensor than `voc`. Here is the state that I see in Zigbee2MQTT:
```
{
    "humidity": 40,
    "linkquality": 232,
    "pm25": 6,
    "temperature": 31,
    "update": {
        "installed_version": 16777233,
        "latest_version": 16777233,
        "state": "idle"
    },
    "voc_index": 283
}
```
In Home Assistant, I see `pm25` in `µg/m³`, and `VOC index` is unitless:
![image](https://github.com/user-attachments/assets/e019e4c1-929c-4e3f-a91e-83fc28857c55)

**Reply by tomaszn** [reply]
If I correctly assume that the sensor inside is SEN54 ([source](https://community.home-assistant.io/t/ikea-vindstyrka-zigbee-air-quality-sensor/549747/)), then the units for tVOC (total VOC) are (should be) ppm ([source](https://sensirion.com/products/catalog/SEN54)).

**Reply by sjorge** [reply]
voc_index is sadly not tVOC and not ppb|ppm|µg/m³, It's a relative offset to the previous 24h, ikea uses it to map to the red/yellow/green an the Up or Down arrow depending on the trend. IIRC 150 is the base line so >150 is worse air quality <150 is better air quality compared to the last 24h average. The further from 150 the better/worse it is.

**Reply by tomaszn** [reply]
Oh, you are right. Details are described [here](https://sensirion.com/media/documents/02232963/6294E043/Info_Note_VOC_Index.pdf). The key sentence is:
> The VOC Index mimics the human nose’s perception of odors
with a relative intensity compared to recent history. The VOC
Index is also sensitive to odorless VOCs, but it cannot
discriminate between them.

**Reply by sjorge** [reply]
A possible improvement would be to map the voc_index value to the color codes perhaps and expose it as an event to home assistant.

**Reply by tomaszn** [reply]
Or even display the trend value as arrows, the same way as visible on the built-in VINDSTYRKA display: 🡲 🡵 🡶.
